src/semant/translate.sml

```diff
@@ -181,12 +193,13 @@ structure Translate : TRANSLATE = struct
                             T.CONST(0))))
     end
 
-  fun for'(lo,hi,body,join) =
-    let val loR = T.TEMP(Temp.newtemp())
+  fun for'(loLoc,lo,hi,body,join) =
+    let val loR = followStaticLink(loLoc)
```

Comment:
Why do you need to follow static links here, shouldn't the loop variable always be at the current frame? Or is this because its simpler to handle the case where it's a reg or a frame variable this way?